src/routes.py

```diff
@@ -0,0 +1,94 @@
+from flask_restplus import Api, Resource, fields
+
+import auth_util
+import email_client
+import sqlalchemy
+from models import db, Token, User
+
+api = Api(version='0.1', title='Quokka API')
+namespace = api.namespace('api')
+
+user_model = api.model('User', {
+    'id': fields.Integer(readOnly=True, description='The user unique identifier'),
+    'email': fields.String(),
+})
+
+create_user_model = api.model('User', {
+    'email': fields.String(),
+    'password': fields.String(),
+})
+
+mint_token_model = api.model('Credentials', {
+    'email': fields.String(),
+    'password': fields.String(),
+})
+
+unauthorized_response_model = api.model('Unauthorized', {
+    'reason': fields.String(),
+})
+
+token_model = api.model('Token', {
+    'token': fields.String(),
+})
+
+@namespace.route('/users/')
+class CreateUser(Resource):
+    """Lets you POST to add new users"""
+
+    @namespace.doc('create_user')
+    @namespace.expect(create_user_model)
+    @namespace.marshal_with(user_model, code=201)
+    def post(self):
+        user_exists_already = User.query.filter_by(email=api.payload['email']).first() is not None
+        user = User(
+            hashed_pw=auth_util.hash_pw(api.payload['password']),
+            email=api.payload['email'])
+        if not user_exists_already:
+          db.session.add(user)
+          db.session.commit()
+        email_client.send_verification_email(user)
+        return user, 201
+
+# TODO add endpoint for resending user verification email
+
+@namespace.route('/verify_email/<int:user_id>/<string:secret>')
+class VerifyEmail(Resource):
+    """Verifies an email address."""
+    @namespace.response(200, 'Success')
+    @namespace.response(401, 'Secret unrecognized')
+    def post(self, user_id, secret):
+        user = User.query.filter_by(id=user_id).first()
+        if user is not None and auth_util.verify_secret(
+                secret, user.email_verification_secret):
+            user.email_verified = True
```

Comment:
I'm not sure I follow. I believe `test_create_user_flow` demonstrates this working correctly.

That said, I've now cherry picked f/schools_fetch, which I believe makes whatever change you're talking about here.